txpool/txpool.go

```diff
@@ -814,7 +814,7 @@ func (p *TxPool) addTx(origin txOrigin, tx *types.Transaction) error {
 
 		slotsFree += slotsRequired(oldTxWithSameNonce) // add old tx slots
 	} else {
-		if account.enqueued.length() == account.maxEnqueued {
+		if account.enqueued.length() == account.maxEnqueued && tx.Nonce != accountNonce {
 			return ErrMaxEnqueuedLimitReached
```

Comment:
Basically, we are going to return the error only in case the max enqueued limit is reached and the nonce-gapped transaction is submitted. I'm wondering do we then even need the `maxEnqueued` parameter in this case, and if not is it safe to remove it (though how to protect from spammers in that case)?